### PR TITLE
Add none_of method

### PR DESCRIPTION
**Pull request by e-cal**

### CODE REVIEW

**Comment by KarishmaDaga** [comment]
hey ethan! this sounds like a great addition to constraint building :)

for a user be able to use ``none_of``, it'll need to be accessible as both a decorator and function (for direct constraint adding) in ``bauhaus.core.constraint.py``

sample: 
```
# decorator
  @constraint.none_of(E)
  @proposition
  class A:
      pass
  
# direct constraint creation
  constraint.add_none_of(E, [propositional variables])
```

I'll link two examples here for you to follow for creating new decorator and direct constraint functions:

decorator: https://github.com/QuMuLab/bauhaus/blob/main/bauhaus/core.py#L389
direct: https://github.com/QuMuLab/bauhaus/blob/main/bauhaus/core.py#L528

also an aside -- I think calling it ``negate_all`` could make the purpose of the constraint more clear, but I don't know if ``none_of`` is a standard name for this type of encoding. cc @haz!

**Comment by haz** [comment]
I guess it mirrors `oneof` to some extent? `negate_all` feels like the name of something that changes objects in place rather than returning a boolean.

All of the other advice is spot on. Also need to make sure it's embedded in the docs and tested.

Final thought -- could we just get away with `~Or(inputs)` instead of the list comprehension?

@KarishmaDaga More PR's coming atcha soon!

**Comment by e-cal** [comment]
@haz 
> Final thought -- could we just get away with `~Or(inputs)` instead of the list comprehension?

That throws the error `TypeError: bad operand type for unary ~: 'And'` (pretty sure you meant `And` not `Or`)

**Comment by haz** [comment]
Bah, silly `python-nnf`. So it'd have to be `Or(inputs).negate()`. `Not` of `ors` is `and` of `nots` (DeMorgan's and all).

**Comment by e-cal** [comment]
Right my bad!

**Comment by e-cal** [comment]
Let me know if I missed anything with the testing. I don't see a script or any references to the testing methods, so I don't know where to run them.

Also, I'm not sure if I should be instantiating the class I added, but if I try to append it to the `objects` list my linter gives me a type compatibility error. I could make a second `objects2` list or instantiate the class separately (`a2 = A2()`), but the `objects` list isn't accessed anywhere so maybe its not necessary?

File: bauhaus/core.py
Comment (KarishmaDaga) [inline bauhaus/core.py:528]: move this function to above the comments on line 524/directly after the implies all function!

**Review by KarishmaDaga — CHANGES_REQUESTED** [review]
great work! other than moving the decorator none_of function up a few lines, everything lgtm :) once that change is done let's merge this pr!

**Review by KarishmaDaga — APPROVED** [review]

**Pull request merged**
